src/festim/stepsize.py

```diff
@@ -67,16 +92,59 @@ def cutback_factor(self, value):
 
         self._cutback_factor = value
 
+    @property
+    def max_stepsize(self):
+        return self._max_stepsize
+
+    @max_stepsize.setter
+    def max_stepsize(self, value):
+        if isinstance(value, float):
+            if value < self.initial_value:
+                raise ValueError(
+                    "maximum stepsize cannot be less than initial stepsize"
+                )
+
+        self._max_stepsize = value
+
+    def get_max_stepsize(self, t):
+        """
+        Returns the maximum stepsize at time t.
+
+        Args:
+            t (float): the current time
+
+        Returns:
+            float or None: the maximum stepsize at time t
+        """
+        if callable(self._max_stepsize):
+            return self._max_stepsize(t)
+        return self._max_stepsize
+
     def modify_value(self, value, nb_iterations, t=None):
         if not self.is_adapt(t):
             return value
 
         if nb_iterations < self.target_nb_iterations:
-            return value * self.growth_factor
+            updated_value = value * self.growth_factor
         elif nb_iterations > self.target_nb_iterations:
-            return value * self.cutback_factor
+            updated_value = value * self.cutback_factor
         else:
-            return value
+            updated_value = value
+
+        max_step = self.get_max_stepsize(t)
+        if max_step:
```

Comment:
Actually I think you can use the `:=` operator for this!!!

```suggestion
        if max_step := self.get_max_stepsize(t):
```